server/course/router.go

```diff
@@ -100,7 +102,7 @@ func createCourse(c *gin.Context) {
 		return
 	}
 
-	course, err := CreateCourse(c, newCourse)
+	course, err := CreateCourse(c, newCourse, userID)
 	if err != nil {
 		handleError(c, http.StatusInternalServerError, err)
```

Comment:
I would not return the plain error here. This may leak information to the enduser/attacker. This applies to most errors in this file here.

I would recommend to `log` the error and return a more generic one